selftests/unit/test_utils_cpu.py

```diff
@@ -319,6 +319,216 @@ def test_x86_64(self):
+cache0          : level=1 type=Data scope=Private size=96K line_size=256 associativity=6
+cache1          : level=1 type=Instruction scope=Private size=64K line_size=256 associativity=4
+cache2          : level=2 type=Data scope=Private size=1024K line_size=256 associativity=8
+cache3          : level=2 type=Instruction scope=Private size=1024K line_size=256 associativity=8
+cache4          : level=3 type=Unified scope=Shared size=49152K line_size=256 associativity=12
+cache5          : level=4 type=Unified scope=Shared size=393216K line_size=256 associativity=24
+processor 0: version = 00,  identification = 3FC047,  machine = 2827
+processor 1: version = 00,  identification = 3FC047,  machine = 2827
+
+cpu number      : 0
+cpu MHz dynamic : 5504
+cpu MHz static  : 5504
+
+cpu number      : 1
+cpu MHz dynamic : 5504
+cpu MHz static  : 5504
+"""
+        with mock.patch('avocado.utils.cpu.open',
+                        return_value=self._get_file_mock(cpu_output)):
+            self.assertEqual(cpu.get_cpu_arch(), "s390")
+
+
+    @unittest.skipUnless(recent_mock(),
+                         "mock library version cannot (easily) patch open()")
+    def test_cpu_arch_arm_v7(self):
+        cpu_output = u"""Processor       : ARMv7 Processor rev 2 (v7l)
+BogoMIPS        : 994.65
+Features        : swp half thumb fastmult vfp edsp thumbee neon vfpv3
+CPU implementer : 0x41
+CPU architecture: 7
+CPU variant     : 0x2
+CPU part        : 0xc08
+CPU revision    : 2
+
+Hardware        : herring
+Revision        : 0034
+Serial          : 3534268a5e0700ec
+"""
+        with mock.patch('avocado.utils.cpu.open',
+                        return_value=self._get_file_mock(cpu_output)):
+            self.assertEqual(cpu.get_cpu_arch(), "arm")
+
+
+    @unittest.skipUnless(recent_mock(),
+                         "mock library version cannot (easily) patch open()")
+    def test_cpu_arch_arm_v8(self):
+        cpu_output = u"""processor       : 0
+BogoMIPS        : 200.00
+Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
+CPU implementer : 0x43
+CPU architecture: 8
+CPU variant     : 0x1
+CPU part        : 0x0a1
+CPU revision    : 1
+"""
+        with mock.patch('avocado.utils.cpu.open',
+                        return_value=self._get_file_mock(cpu_output)):
+            self.assertEqual(cpu.get_cpu_arch(), "arm")
+
+
+    @unittest.skipUnless(recent_mock(),
+                         "mock library version cannot (easily) patch open()")
+    def test_cpu_arch_risc_v(self):
+        cpu_output = u"""hart	: 1
+isa	: rv64imafdc
+mmu	: sv39
+uarch	: sifive,rocket0
+"""
+        with mock.patch('avocado.utils.cpu.open',
+                        return_value=self._get_file_mock(cpu_output)):
+            self.assertEqual(cpu.get_cpu_arch(), "risc-v")
```

Comment:
I think I'd prefer `riscv` because that is how it's called in kernel, but elsewhere I saw "riscv64" and somewhere the "risc-v". I asked Richard about his opinion and let you know about his opinion.